src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -2310,28 +2321,24 @@ FixedwingPositionControl::Run()
 
 		// handle estimator reset events. we only adjust setpoins for manual modes
 		if (_control_mode.flag_control_manual_enabled) {
-			if (_control_mode.flag_control_altitude_enabled && _local_pos.vz_reset_counter != _alt_reset_counter) {
+			if (_control_mode.flag_control_altitude_enabled && _local_pos.vz_reset_counter != _vz_reset_counter) {
 				// make TECS accept step in altitude and demanded altitude
 				_tecs.handle_alt_step(_current_altitude, -_local_pos.vz);
 			}
 
 			// adjust navigation waypoints in position control mode
 			if (_control_mode.flag_control_altitude_enabled && _control_mode.flag_control_velocity_enabled
-			    && _local_pos.vxy_reset_counter != _pos_reset_counter) {
+			    && _local_pos.vxy_reset_counter != _vxy_reset_counter) {
 
 				// reset heading hold flag, which will re-initialise position control
 				_hdg_hold_enabled = false;
 			}
 		}
 
-		// update the reset counters in any case
-		_alt_reset_counter = _local_pos.vz_reset_counter;
-		_pos_reset_counter = _local_pos.vxy_reset_counter;
```

Comment:
...

Reply:
I dont know what you're referring to. I changed the naming to match the actual reset counter and moved it to the end of the function.
Before moving it, the condition `_local_pos.vxy_reset_counter != _pos_reset_counter` ([here](https://github.com/PX4/PX4-Autopilot/pull/23446/files#diff-7e36fecc9802c57f75d7073fcd950b48b8228356d0e7cf65dcf2ce4c23bd563fL2334)) was never true since `_pos_reset_counter` was updated prior to the comparison.